src/java_bytecode/java_types.h

```diff
@@ -120,10 +120,14 @@ class java_generic_parametert:public reference_typet
     return const_cast<type_variablet &>(type_variables().front());
   }
 
+  const irep_idt get_name() const
+  {
+    return type_variable().get_identifier();
+  }
+
   const std::string get_parameter_class_name() const
   {
-    const std::string &parameter_name =
-      type_variable().get_identifier().c_str();
+    const std::string &parameter_name = get_name().c_str();
```

Comment:
why not `id2string` instead of `c_str` here?

Reply:
This function is actually not used anymore (was part of the old implementation for specialization) so will remove.

Reply:
This still seems to be here? Perhaps removed in a later commit